frontend/actions/src/main/kotlin/com/virtuslab/gitmachete/frontend/actions/dialogs/SlideInDialog.kt

```diff
@@ -42,6 +43,9 @@ class SlideInDialog(
 
   private val rootNames = branchLayout.rootEntries.map { it.name }
 
+  private val branchToAnnotation = branchLayout.rootEntries.flatMap(::collectEntries).map { it.name to it.customAnnotation.orEmpty() }.toMap()
```

Comment:
I am creating a cache in the very beginning instead of calling `getEntryByName` every time one modifies the branch name field